tensorflow/compiler/jit/xla_compilation_cache.h

```diff
@@ -189,6 +214,30 @@ class XlaCompilationCache : public ResourceBase {
   absl::flat_hash_map<string, ClusterCompileStats> cluster_compile_stats_
       TF_GUARDED_BY(cluster_compile_stats_mu_);
 
+  // KEEP THIS STRUCT LAST!!! It guarantees that the ~ThreadPool destructor is
+  // called first when the cache is destroyed. This guarentees the cache to be
+  // alive when asynchronous compilations are still happening.
```

Comment:
Can we avoid the need to be careful here by putting this in a `std::unique_ptr<AsyncCompilation>` and resetting that in the `XlaCompilationCache` destructor?  That way the threadpool is always destroyed first irrespective of order.

I'm worried that folks are going to add more fields anyway since typically people don't read the comments on the entire class before adding fields.

Reply:
done, but for ThreadPool only (there's precedent for that already)